pepper-apis/src/main/java/org/broadinstitute/ddp/filter/RateLimitFilter.java

```diff
@@ -0,0 +1,36 @@
+package org.broadinstitute.ddp.filter;
+
+import java.util.Map;
+import javax.servlet.ServletException;
+
+import org.broadinstitute.ddp.exception.DDPException;
+import org.eclipse.jetty.servlets.DoSFilter;
+import spark.Filter;
+import spark.Request;
+import spark.Response;
+
+public class RateLimitFilter implements Filter {
+
+    private final int maxRequestsPerSecond;
+    private final int burst;
+
+    public RateLimitFilter(int maxRequestsPerSecond, int burst) {
+        this.maxRequestsPerSecond = maxRequestsPerSecond;
+        this.burst = burst;
+    }
+
+    @Override
+    public void handle(Request request, Response response) throws Exception {
+        DoSFilter rateLimitFilter = new DoSFilter();
+        rateLimitFilter.setEnabled(true);
+        try {
+            rateLimitFilter.init(new FilterConfig("DOSFilter", Map.of("delayMs", "-1",
+                    "maxRequestsPerSec", Integer.toString(maxRequestsPerSecond),
+                    "throttledRequests", Integer.toString(burst),
+                    "tooManyCode", "503")));
```

Comment:
If possible, I'll suggest using `429` here.

Reply:
PR here is not showing the latest change, where `tooManyCode` is removed (and thus falls back to the default `429`).